### DISCUSSION TITLE
Central Car Integration

### DISCUSSION THREAD

**Discussion by cdnninja** [home-assistant/architecture#708] [General]
@fuatakgun and I were chatting awhile back as we worked on the Kia UVO / Blue Link integration that many cars use the same sort of approach.  They all contain things like remote start, door sensors, fuel levels etc.  While the sensors each vehicle supports may be different on the edge cases the core is often the same.    With some rewrite work that  @fuatakgun did it wouldn't be hard to rename https://github.com/home-assistant/core/pull/62451 into a integration meant for future car brands too. 

Each brand would need to have someone maintain the python library against the same standard methods but with the current design of the Kia / Bluelink API we believe it is flexible enough to work for many brands and regions.

Would need to get other car code owners involved - not sure if this makes things easier or harder.   Part of my thinking is it will keep cars consistent in home assistant.

**Comment by fuatakgun** [comment]
As an additional point, we can take it further and discuss about device templates.

As an example a vehicle device template can include core entities (sensors) in it and all integration owners working on a smart vehicle integration can use this template to define all entities. Moreover, it should be extendible with additional sensors do any further customizations can be done on top.

İ just put very basic examples below and i am open to discuss further.

Example Vehicle device template;
- fuel level - sensor
- lock - lock
- 12v battery level - sensor
- door(s) open - binary sensor
- trunk open - binary sensor

Example Smart Vacuum Cleaner;
- battery level - sensor
- running - binary sensor
- last run - sensor
- next scheduled run - sensor
- last clean - sensor

**Reply by Jc2k** [reply]
Given the smart vacuum example, this might be better as a seperate topic?

**Comment by Jc2k** [comment]
Could you elaborate on what the problem is and what HA needs to do to solve it?

The idea as stated originally seems to be that you can move the complexity out of HA and into a python package that supports multiple car brands? Then one integration would support multiple cars via this single new library? Is that an accurate interpretation?

Without knowing much about smart cars, it's not clear that there would be much code sharing between brands? Unless they all licence the same smart cart platform? So what are the advantages you see?

**Reply by cdnninja** [reply]
Sorry for not replying sooner. 

 That's correct. What I have found is a good bit of work has been done to dial in the home assistant side of the code.  Cars in general have large overlap of what they do.  Just need to abstract it.  Our current integration already supports 4 different APIs. For a total of 6 brands/regions. 

We find a few things that could be standardized and improved in ha itself that would assist this.

**Reply by ReneNulschDE** [reply]
@cdnninja: ping me on https://community.home-assistant.io/t/mercedes-me-component/41911 in case this topic gets more focus. I'm maintaining the Mercedes Benz integration for several years now and moved it from the general HA-release (0.6.x) to a custom component as I use some unofficial APIs.

**Comment by frenck** [comment]
Oh hi there 👋 

I'm going through some older proposals and bumped into this one. We are current working don't the concept of actual devices (aka as device types). A car would (potentially) be such a thing. We are at this point not planning on adding a car entity type.

For that reason, I'll go ahead and close this proposal.

../Frenck
